src/main/java/com/github/onsdigital/dp/image/api/client/ImageAPIClient.java

```diff
@@ -0,0 +1,163 @@
+     *
+     * @param collectionID A string containing a required collectionID
+     * @return An {@link Images} object containing a list of Image objects
+     * @throws IOException
+     * @throws ImageAPIException
+     */
+    @Override
+    public Images getImagesWithCollectionId(String collectionID) throws IOException, ImageAPIException {
+
+        validateCollectionID(collectionID);
+
+        String path = "/images?collection_id=" + collectionID;
+        URI uri = imageAPIURL.resolve(path);
+
+        HttpGet req = new HttpGet(uri);
+        req.addHeader(serviceTokenHeaderName, serviceAuthToken);
+
+        try (CloseableHttpResponse resp = executeRequest(req)) {
+            int statusCode = resp.getStatusLine().getStatusCode();
+
+            switch (statusCode) {
+                case HttpStatus.SC_OK:
+                    return parseResponseBody(resp, Images.class);
+                default:
+                    throw new ImageAPIException(formatErrResponse(req, resp), statusCode);
+            }
+        }
+    }
+
+    /**
+     * Publish an image by calling the POST /images/{id}/publish endpoint
+     *
+     * @param imageId A string containing the id of a specific image to publish
+     * @throws IOException
+     * @throws ImageAPIException
+     */
+    @Override
+    public void publishImage(String imageId) throws IOException, ImageAPIException {
+
+        validateImageID(imageId);
+
+        String path = "/images/" + imageId + "/publish";
+        URI uri = imageAPIURL.resolve(path);
+
+        HttpPost req = new HttpPost(uri);
+        req.addHeader(serviceTokenHeaderName, serviceAuthToken);
+
+        try (CloseableHttpResponse resp = executeRequest(req)) {
+            int statusCode = resp.getStatusLine().getStatusCode();
+
+            switch (statusCode) {
+                case HttpStatus.SC_NO_CONTENT:
+                    return;
+                default:
+                    throw new ImageAPIException(formatErrResponse(req, resp), statusCode);
+            }
+        }
+    }
+
+    private void validateCollectionID(String collectionID) {
+        Args.check(StringUtils.isNotEmpty(collectionID), "an collection id must be provided.");
+    }
+
+    private void validateImageID(String imageID) {
+        Args.check(StringUtils.isNotEmpty(imageID), "an image id must be provided.");
+    }
+
+    private <T> T parseResponseBody(CloseableHttpResponse response, Class<T> type) throws IOException {
+        HttpEntity entity = response.getEntity();
+        String responseString = EntityUtils.toString(entity);
+        return json.readValue(responseString, type);
+    }
+
+    private String formatErrResponse(HttpRequestBase httpRequest, CloseableHttpResponse response) {
+        return String.format("the image api returned a %s response for %s",
+                response.getStatusLine().getStatusCode(),
+                httpRequest.getURI());
```

Comment:
Is there value in including the expected status code as well? My thinking is that it will provide instant clarity as to the expected behaviour without having to consult the code.